### ISSUE TITLE
Function keys i.e. F5 and F6 input random characters on windows console

### ISSUE THREAD

**Issue by musm** [JuliaLang/julia#28799]
On julia v1 and earlier version, hitting various function keys in the REPL outputs junk on Windows. This is reproducible.

E.g.: hitting the F5 key outputs  E   ,  F4 outputs D , F3 outputs C, F2 outputs B, and F1 outputs A

**Comment by woodwm** [comment]
I can reproduce this with Julia 0.7 on windows X64. It may be an Easter egg. 😂

In addition, press `Esc` then `F1`, you will get `[[A` and similar output for the above mentioned function keys `F2~F5`. If you press `Esc` then `F6`, you will get `[17~` and similar output for `F7~F12` and `Insert`, `Delete`, `Home`, `End`.

**Comment by musm** [comment]
What's responsible for handling keyboard input?

**Comment by digital-carver** [comment]
This issue still persists as of Julia 1.12 (tested on Windows Terminal and Wezterm). Pressing `F1` to `F5` inputs charactes `A` to `E` correspondingly. (The title here seems wrong btw, `F6` does not have this issue, nor do `F7`-`F12`.)

This interferes with the working of the new `F1` for paged help [functionality in TerminalPager](https://github.com/ronisbr/TerminalPager.jl/releases/tag/v0.6.5), as a concrete example. Pressing `Esc` then `O` then `P` (so inputting `\eOP`, the escape code for `F1`, directly) does bring up the paged help, so the issue seems to be that `F1` isn't being seen as `\eOP` by the REPL input system.

**Comment by matthias314** [comment]
@digital-carver What key codes does the Windows Terminal (or whichever terminal emulator you are using) produce for function keys? Are they documented somewhere?

**Comment by digital-carver** [comment]
tl;dr: This is possibly/likely solved in Julia 1.13 with https://github.com/JuliaLang/julia/pull/59825 , though it would be good to have confirmation from another Windows user on that before we close this. 

For future record: The clue came from running Julia within WSL in the same terminal apps - there, the function keys are processed properly, and for eg. `F1` for TerminalPager works (on Julia under the WSL Linux). Also, looking into keycodes as @matthias314 suggested, both OSs are able to receive `\eOP` for `F1` outside Julia in the shell CLI. So it's not the terminal per se that's the issue, it's the Julia REPL - which led me to look into the [only Windows-specific part of the REPL input code](https://github.com/JuliaLang/julia/blob/78e95d2e67f2460cc48d13263f1d3709c7264c17/stdlib/REPL/src/LineEdit.jl#L1738) and the one flag not set there: `ENABLE_VIRTUAL_TERMINAL_INPUT` (as mentioned in https://github.com/JuliaLang/julia/issues/53763 ). That took me to https://github.com/JuliaLang/julia/pull/59825 where that flag was switched on to support bracketed paste mode - and happily, that seems to have fixed this issue as well. 

Now, with Julia 1.13 beta, the `F1` mapping from TerminalPager.jl works on Windows too. For more testing I used this quick-and-dirty (AI assisted) script:
<details><summary>functionkeytester.jl</summary>

```
# Bind simple test handlers for F1..F12 on the main prompt (mode 1).

function install_fn_tests(; mode_index::Int=1)
    km = Base.active_repl.interface.modes[mode_index].keymap_dict

    # helper: walk existing trie, set leaf
    setseq!(km, seq::AbstractString, f) = begin
        node = km
        cs = collect(String(seq))
        for i = 1:length(cs)-1
            if !haskey(node, cs[i])
                node[cs[i]] = Dict{Char, Any}()
            end
            node = node[cs[i]] 
        end
        haskey(node, cs[end]) && @warn("overwriting keymap value for $seq")
        node[cs[end]] = f
        nothing
    end

    # sequences to bind (common SS3 + tilde family)
    seqs = [
        ["\eOP",  "\e[11~"],
        ["\eOQ",  "\e[12~"],
        ["\eOR",  "\e[13~"],
        ["\eOS",  "\e[14~"],
        ["\e[15~"],
        ["\e[17~"],
        ["\e[18~"],
        ["\e[19~"],
        ["\e[20~"],
        ["\e[21~"],
        ["\e[23~"],
        ["\e[24~"],
    ]

    for (n, vs) in pairs(seqs)
        for seq in vs
            setseq!(km, seq, (s, args...) -> (println("F$n received as $(repr(seq))"); nothing))
        end
    end
end
```

</details> 

After `include`ing this on 1.12, pressing the function keys in order leads to 
```
julia> install_fn_tests()

julia> ABCDEF6 received as "\e[17~"
F7 received as "\e[18~"
F8 received as "\e[19~"
F9 received as "\e[20~"
F10 received as "\e[21~"
F12 received as "\e[24~"
``` 
i.e. `F1` to `F5` print `A` to `E` and don't trigger the key mappings. 

On 1.13, 
```
julia> install_fn_tests()

julia> F1 received as "\eOP"
F2 received as "\eOQ"
F3 received as "\eOR"
F4 received as "\eOS"
F5 received as "\e[15~"
F6 received as "\e[17~"
F7 received as "\e[18~"
F8 received as "\e[19~"
F9 received as "\e[20~"
F10 received as "\e[21~"
F12 received as "\e[24~"
```

So the issue seems fixed, but since it's a UI issue that's hard to write tests for, it would be nice to get some practical confirmation, perhaps on a couple of other terminals on Windows, from other users.
